### PR TITLE
[ez] Windows log printing + save successful test logs

### PR DESCRIPTION
**Pull request by clee2000**
when doing print(f.read().decode etc etc) it prints an extra new line, so manually splitlines and strip to see if that helps

My guess is windows line ending differences

Also always save log file regardless of success or failure

See https://hud.pytorch.org/pytorch/pytorch/commit/476b81a9bfcc765125984ad521e5e40cdbdbf098 for what it looks like now

Swapped to opening in text mode instead of binary, seems to be ok now.

42483193bf024983060a234dc0262f4840aef4b8 for example

### CODE REVIEW

File: test/run_test.py [hunk 1]
```diff
@@ -1029,14 +1029,14 @@ def handle_log_file(
     with open(file_path, "rb") as f:
         full_text = f.read().decode("utf-8", errors="ignore")
 
+    new_file = "test/test-reports/" + sanitize_file_name(
```
Comment (huydhn) [inline test/run_test.py:1032]: So, this is not really about Windows log printing, but this is to persist all the failure logs in the test reports, right?  Like when `keep-going` is used.

Comment (clee2000) [reply test/run_test.py:1032]: yeah, usually it deletes logs if it gets printed, this part is to make sure that the logs are always saved

File: test/run_test.py [hunk 2]
```diff
@@ -1045,11 +1045,12 @@ def handle_log_file(
...
         return
-    # otherwise: print entire file and then remove it
-    print_to_stderr(f"\nPRINTING LOG FILE of {test} ({file_path})")
-    print_to_stderr(full_text)
-    print_to_stderr(f"FINISHED PRINTING LOG FILE of {test} ({file_path})\n")
-    os.remove(file_path)
+
+    # otherwise: print entire file
+    print_to_stderr(f"\nPRINTING LOG FILE of {test} ({new_file})")
+    for line in full_text.splitlines():
```
Comment (huydhn) [inline test/run_test.py:1051]: This's probably ok.  On the other hand, I think if we could open in text mode, i.e. `with open(file_path, "r") as f` then the `\r\n` newlines on Windows will be automatically handled I think.  I vaguely remember that opening in binary mode then decoding with ignore error is to skip over bad characters.  Then there is also `errors=ignore` when opening a file, I wonder why we don't use that.

**Review by huydhn — APPROVED** [review]
LGTM!  May be trying to see if `with open(file_path, mode="rb", errors="ignore") as f` works as an alternative.

**Comment by clee2000** [comment]
> LGTM! May be trying to see if `with open(file_path, mode="rb", errors="ignore") as f` works as an alternative.

Swapped to this, except just "r" since the "b" is for binary, which seems to solve the problem

**Pull request closed without merging**